AlphaWallet/UI/AddressTextField.swift

```diff
@@ -405,27 +296,39 @@ extension AddressTextField: UITextFieldDelegate {
     }
 
     func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
-        clearAddressFromResolvingEnsName()
+        ensAddressLabel.clear()
+
         guard delegate != nil else { return true }
-        let newValue = (self.textField.text as NSString?)?.replacingCharacters(in: range, with: string)
-        if let newValue = newValue {
-            if CryptoAddressValidator.isValidAddress(newValue) {
-                DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
-                    //Retain self because it's still useful to resolve and cache even if not used immediately
-                    self.queueEnsResolution(ofValue: newValue)
-                }
-            } else if newValue.isPossibleEnsName {
-                DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
-                    //Retain self because it's still useful to resolve and cache even if not used immediately
-                    self.queueAddressResolution(ofValue: newValue)
-                }
-            }
+        let newValue = ((textField.text ?? "") as NSString).replacingCharacters(in: range, with: string)
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
+            self.ensAddressResolve(value: newValue)
         }
 
-        informDelegateDidChange(to: newValue ?? "")
+        informDelegateDidChange(to: newValue)
         return true
     }
 
+    private func ensAddressResolve(value: String) {
+        ensAddressLabel.resolve(value) { [weak self] resolvation in
+            guard let strongSelf = self, value == strongSelf.textField.text?.trimmed else {
+                return
+            }
+
+            switch resolvation {
```

Comment:
Seems like this switch can be extracted into a function. It's quite similar in `pasteAction()`, except for an error being displayed?